graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerAdapter.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.indexer.fieldtypes;
+
+import com.codahale.metrics.Timer;
+
+import java.util.Map;
+import java.util.Optional;
+
+public interface IndexFieldTypePollerAdapter {
+    Optional<Map<String, String>> pollIndex(String indexName, Timer pollTimer);
```

Comment:
Why not use `IndexFieldTypesDTO` here instead of just `Map<String, String>`? I think it would communicate the method's purpose more clearly.